materialize-snowflake/snowflake.go

```diff
@@ -452,14 +429,37 @@ func (d *transactor) Store(it *m.StoreIterator) (m.StartCommitFunc, error) {
 					Table:     b.target.Identifier,
 					Query:     mergeIntoQuery,
 					StagedDir: dir,
+					Version:   d.version,
 				}
 			}
 		} else if b.pipeName != "" {
+			var pipeNameParts = strings.Split(b.pipeName, ".")
+
+			// Check to see if a pipe for this version already exists
+			var query = fmt.Sprintf("SHOW PIPES LIKE '%s';", pipeNameParts[len(pipeNameParts)-1])
+			rows, err := d.db.QueryContext(ctx, query)
+			if err != nil {
+				return nil, fmt.Errorf("finding pipe %q: %w", b.pipeName, err)
+			}
+			var pipeExists = rows.Next()
+
+			// Only create the pipe if it doesn't exist. Since the pipe name is versioned by the spec
+			// it means if the spec has been updated, we will end up creating a new pipe
+			if !pipeExists {
+				log.WithField("name", b.pipeName).Info("store: creating pipe")
+				if createPipe, err := RenderTableShardVersionTemplate(b.target, d._range.KeyBegin, d.version, d.templates.createPipe); err != nil {
```

Comment:
There was some concern about `CREATE PIPE` being eventually consistent as well. Since we will ignore "Pipe not found" errors in Acknowledge now, after getting the success response for creating the pipe should we query `SHOW PIPES` to make sure it does show up as existing? For materializations tailing a collection that don't have an update delay set, the time between creating the pipe and doing the acknowledge may be very short.